app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php

```diff
@@ -218,8 +218,22 @@ public function unserializeIntArray($serializedData) {
      *
      * @return  array the converted array of strings in PHP format
      */
-    public function unserializeStringArray($serializedData) {
+    public function unserializeStringArray($serializedData)
+    {
         preg_match_all('/i:\d+;s:\d+:"([^"]*)";/', $serializedData, $convertedArray);
         return $convertedArray[1];
     }
+
+    /***************************************************
+     * Switch Helpers below
+     ***************************************************/
+    public function isSwitchSampleSwitchEnabled()
+    {
+        return Mage::getSingleton("boltpay/featureSwitch")->isSwitchEnabled(Bolt_Boltpay_Model_FeatureSwitch::SAMPLE_SWITCH_NAME);
+    }
+
+    public function isSwitchBoltEnabled()
```

Comment:
At this point, we are abstracting away the underlying switch implementation.  Here is is more desirable to call the method `isBoltEnabled`.

For the other switch, "Sample Switch", I still maintain that it is unusual to be adding example, or demo code in the plugin release.  If it stays, then its logical name would be `isSampleSwitchEnabled`

Reply:
isBoltEnabled is a confusing name because we can have a helper that checks if bolt enabled in bolt settings.
Since we moved helper from feature switches class we need to save "feature switch" in the helper name.
In another hand, we no need to abstract from switch implementation. When we use helpers in our code, we should remember that it's a very specific way, in most cases this feature switch call will be removed after we roll out the new feature.

Reply:
Look at it like this.  In this case, is not about whether a feature switch or setting is enabled.  It is about whether overall if the Bolt is enabled or not.  We abstract away the internal details.

So, in pseudo code, at the helper level we would use:

```
function isBoltEnabled() {
    return isSwitchEnabled && isSettingEnabled;
}
```
And the complexity of the logic could change overtime.  

If we are only concerned about the value of a specific switch, then we just directly use the model method.
 
```
Mage::getSingleton("boltpay/featureSwitch")->isSwitchEnabled(Bolt_Boltpay_Model_FeatureSwitch::BOLT_ENABLED_SWITCH_NAME);
```

Reply:
@vitaliyreznikov 
At any rate, let's move forward with this.  Refactoring can be done at any time.  My largest outstanding question/request was regarding something to test the distribution of our hashing algorithm.  You've mentioned that you have already tested it.  It would be great to add it to our unit test.

Reply:
Sounds good. I'll create new PR to start using this M1_BOLT_ENABLED feature, will discuss method name in this PR

Reply:
@vitaliyreznikov Hopefully you won't mind me giving it a try.  If you don't like my suggestion, you can always present an alternative.